### PR TITLE
Allow functioning on Bash for Windows. Fixes #6

### LINKED ISSUE #6: Code completion should work in Bash on Windows
Related to #2, if you are actually running a Bash shell on Windows, the autocomplete still exits saying it's not compatible with Windows. It should work fine with the Bash shell, so I will confirm this and tweak the OS check to allow running in Bash on Windows.

### PR DESCRIPTION
**Pull request by dprothero**

### CODE REVIEW

File: test/base.test.ts [hunk 1]
```diff
@@ -17,16 +17,38 @@ const config = new Config({root})
...
+
   it('#errorIfWindows', async () => {
+    let lastError
+    cmd.config.windows = true
+    try {
+      cmd.errorIfWindows()
+    } catch (e) {
+      lastError = e
+    }
+    expect(lastError.message).to.eq('Autocomplete is not currently supported in Windows')
```
Comment (childish-sambino) [inline test/base.test.ts]: Little bit cleaner using `to.throw`: https://www.chaijs.com/api/bdd/#method_throw

File: src/commands/autocomplete/index.ts [hunk 2]
```diff
@@ -24,7 +24,7 @@ export default class Index extends AutocompleteBase {
 
   async run() {
     const {args, flags} = this.parse(Index)
-    const shell = args.shell || this.config.shell
+    const shell = args.shell || this.convertWindowsBash(this.config.shell)
     this.errorIfNotSupportedShell(shell)
```
Comment (childish-sambino) [inline src/commands/autocomplete/index.ts]: `errorIfNotSupportedShell` handles a missing `shell`, but `convertWindowsBash` does not (i.e., what if `this.config.shell` is `null/undefined`?).

And should `convertWindowsBash` wrap both shell options?
```
const shell = this.convertWindowsBash(args.shell || this.config.shell)
```

Comment (dprothero) [reply src/commands/autocomplete/index.ts]: It _could_ wrap both, but it wouldn't do anything. The `args.shell` is only going to be "bash" or "zsh" as allowed by the args to the command, whereas `this.config.shell` has a full path to the shell EXE on Windows (which we want to strip out and convert to just "bash", if it's a path to bash.exe).

Comment (dprothero) [reply src/commands/autocomplete/index.ts]: Regarding the missing `shell` scenario. It's not possible because this is TypeScript and `config.shell` is defined as a `string` and the parameter that `convertWindowsBash` accepts is only type string. I can't even write a test that passes undefined to `convertWindowsBash` because the compiler won't allow it.

I will add a test case for an empty string.

**Comment by dprothero** [comment]
FYI, in testing this, it's not working quite right on Bash on Windows... Will need to look into mapping the paths from Windows style paths (e.g. `C:\Users\<user>\AppData\Local\@twilio\cli\autocomplete\functions\bash\twilio.bash`) to bash-style paths (e.g. `~/AppData/Local/@twilio/cli/autocomplete/functions/bash/twilio.bash`)

File: test/base.test.ts [hunk 3]
```diff
@@ -17,16 +17,38 @@ const config = new Config({root})
...
+    let lastError
+    cmd.config.windows = true
+    cmd.config.shell = 'C:\\bin\\bash.exe'
     try {
       cmd.errorIfWindows()
     } catch (e) {
-      expect(e.message).to.eq('Autocomplete is not currently supported in Windows')
+      lastError = e
     }
+    expect(lastError).to.eq(undefined)
```
Comment (RasPhilCo) [inline test/base.test.ts]: We could use `to.not.throw` here.

**Comment by RasPhilCo** [comment]
hey @dprothero! I think a colleague may take a crack at finishing this PR.

**Comment by williamruemmele-sf** [comment]
Hi @dprothero I see your comment about an issue with '\' vs '/'. I may have a solution but I want to test it. Could you give me some context or steps to reproduce the issue you were seeing with using '\'? Thanks

**Comment by RasPhilCo** [comment]
Cleaning up - if you think this PR is still valid, reopen it. Thanks!
